ssl/t1_trce.c

```diff
@@ -1493,11 +1497,17 @@ void SSL_trace(int write_p, int version, int content_type,
     switch (content_type) {
     case SSL3_RT_HEADER:
         {
-            int hvers = msg[1] << 8 | msg[2];
+            int hvers;
+            /* avoid overlapping with length at the end of buffer */
+            if (msglen < 5)
+                break;
```

Comment:
If you consider overall purpose of this function, you'll have to conclude that it's debugging. And debugging by examining output, not single-stepping. Question is if just breaking (and consequently omitting empty line) is something that would help debugging? Wouldn't user appreciate to note bad things happening (just as much as good ones)? And yes, if taken to the logical conclusion you'd also have to verify that encoded payload length matches (or doesn't exceed?) msglen. Once again, wholeheartedly...

[On *side* note there seems to be inconsistency on how lines are terminated. They all end with \n, except for two...]

Reply:
Did I add the right debug messages in the next patch ?

Reply:
I don't think that you've got indentation right. I mean you used 4, but I don't think it's right. Though to be honest I don't actually know what's right, but it looks like it would be appropriate to align it with what applies in `case` in question, and it seems to be ... 0. But it's kind of odd that you can get "received" followed by "too short" in DTLS case. I'd say that following would be more appropriate:
```
int is_dtls = SSL_IS_DTLS(ssl);

BIO_puts(bio, write_p ? "Sent" : "Received");
if (msglen < is_dtls ? 13 : 5) {
    ssl_print_hex(bio, 1, "too short message", msg, msglen);
    break;
}
...
if (is_dtls) ...
```